VAMobile/src/api/contactInformation/updateAddress.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+
+import { Params as APIParams, EditResponseData, post, put } from 'store/api'
+import { AddressData } from 'api/types'
+import { UserAnalytics } from 'constants/analytics'
+import { contactInformationKeys } from './queryKeys'
+import { isErrorObject } from 'utils/common'
+import { logNonFatalErrorToFirebase, setAnalyticsUserProperty } from 'utils/analytics'
+
+/**
+ * Creates or updates a user's address depending on whether an `id` field is present
+ */
+const updateAddress = async (addressData: AddressData) => {
+  const endpoint = '/v0/user/addresses'
+
+  if (!addressData.id) {
+    return post<EditResponseData>(endpoint, addressData as unknown as APIParams)
+  }
+
+  return put<EditResponseData>(endpoint, addressData as unknown as APIParams)
+}
+
+/**
+ * Returns a mutation for updating an address
+ */
+export const useUpdateAddress = () => {
+  const queryClient = useQueryClient()
+
+  return useMutation({
+    mutationFn: updateAddress,
+    mutationKey: ['updateAddress'],
+    onSuccess: async () => {
+      await setAnalyticsUserProperty(UserAnalytics.vama_uses_profile())
+      queryClient.invalidateQueries({ queryKey: contactInformationKeys.contactInformation })
+    },
+    onError: async (error) => {
+      if (isErrorObject(error)) {
+        logNonFatalErrorToFirebase(error, 'updateAddress: Service error')
+      }
+    },
+  })
```

Comment:
I made a single mutation for both creating and updating an address, since the code is basically identical. In contrast, I made separate mutations for [creating](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733/files#diff-bfe1e2437d52a629b2fdbc0e1845ee616c779ba5416f05f02bb98593b6319f9eR1) and [updating](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733/files#diff-e46a82603c8f50a860943a1931db8cf6262def93a6f0b80acef6e78f5bb0978bR1) phone numbers. I wanted to show what both approaches look like and see if there's a strong preference for which pattern we should adapt. The pros about using one mutation for both creating and updating is fewer lines of code. On the other hand, we lose the ability to differentiate between when the entity is being updated or created, since there would only be one `isLoading` status for the single mutation. Also, semantically, the naming of a single mutation hook might not be as clear to what it does. You can see both approaches in action in the [EditAddressScreen](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733/files#diff-5289db0ef9e10d828712591d88415c37849600f4bed6b3b0a6d48ee0970b182aR281) and [EditPhoneNumberScreen](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733/files#diff-8d68d775f5241ccbc8a26649b596afdf7732e575d7129ddd5c42cf33bbb97402R109-R130) files

Reply:
I personally like having them be in the same mutation as opposed to separate mutation files